src/app/modules/webhook/webhook.service.ts

```diff
@@ -97,81 +111,91 @@ export const sendWebhook = (
     signature,
   }
 
-  return ResultAsync.fromPromise(validateWebhookUrl(webhookUrl), (error) => {
+  return ResultAsync.fromPromise(Bluebird.props(signedUrlPromises), (error) => {
     logger.error({
-      message: 'Webhook URL failed validation',
+      message: 'S3 attachment presigned URL generation failed',
       meta: logMeta,
       error,
     })
-    return error instanceof WebhookValidationError
-      ? error
-      : new WebhookValidationError()
-  })
-    .andThen(() =>
-      ResultAsync.fromPromise(
-        axios.post<unknown>(webhookUrl, submissionWebhookView, {
-          headers: {
-            'X-FormSG-Signature': formsgSdk.webhooks.constructHeader({
-              epoch: now,
-              submissionId,
-              formId,
-              signature,
-            }),
-          },
-          maxRedirects: 0,
-        }),
-        (error) => {
-          logger.error({
-            message: 'Webhook POST failed',
-            meta: {
-              ...logMeta,
-              isAxiosError: axios.isAxiosError(error),
-              status: get(error, 'response.status'),
-            },
-            error,
-          })
-          if (axios.isAxiosError(error)) {
-            return new WebhookFailedWithAxiosError(error)
-          }
-          return new WebhookFailedWithUnknownError(error)
-        },
-      ),
-    )
-    .map((response) => {
-      // Capture response for logging purposes
-      logger.info({
-        message: 'Webhook POST succeeded',
-        meta: {
-          ...logMeta,
-          status: get(response, 'status'),
-        },
+    return new WebhookFailedWithUnknownError(error)
+  }).andThen((signedUrls) => {
+    submissionWebhookView.data.attachmentDownloadUrls = signedUrls
```

Comment:
So that was what I was thinking originally when I suggesting putting the generation in `getWebhook` so it returns the full S3 presigned URL. Should I put it there or create a new function to convert what is returned by `getWebhook`?

Reply:
I think the latter! ideally `getWebhookView` should only return some view of the data, and we should keep application logic in the service

Reply:
Cool, split it into a separate function.